### PR TITLE
Support for transforming a Nadel instance

### PR DESCRIPTION
**Pull request by bbakerman**
After PR feedback I made the  the SchemaTransformHook take extra parameters

Also fixed up the engine factory so it is lazy and only calls the deault if its null on build

### CODE REVIEW

**Review by temaEmelyan — APPROVED** [review]

**Review by andimarek — CHANGES_REQUESTED** [review]
I don't thinks this is the right way as this bypasses a lot of init code. The new engine also relies on the overall schema. Not sure what happens if we change it.

**Review by andimarek — CHANGES_REQUESTED** [review]
small thing around asserts

File: api/src/main/java/graphql/nadel/Nadel.java [hunk 1]
```diff
@@ -442,6 +452,16 @@ public Nadel build() {
...
                     schemaTransformationHook);
+
+            NadelExecutionEngine executionEngine;
+            if (engineFactory == null) {
+                executionEngine = buildDefaultEngine(nadelStep1);
+            } else {
+                executionEngine = engineFactory.create(nadelStep1);
+            }
+
+            Objects.requireNonNull(executionEngine, "The engine factory must return an engine instance");
```
[the remark this answers is not in the document]
Comment (bbakerman) [reply api/src/main/java/graphql/nadel/Nadel.java]: done

**Review by bbakerman — COMMENTED** [review]
done

**Pull request merged**